kbase-extension/static/kbase/js/common/cellComponents/paramsWidget.js

```diff
@@ -0,0 +1,551 @@
+            const removeClass = (settings.showAdvanced ? 'advanced-parameter-hidden' : 'advanced-parameter-showing');
+            const addClass = (settings.showAdvanced ? 'advanced-parameter-showing' : 'advanced-parameter-hidden');
+
+            for (const [key, entry] of Object.entries(advancedInputs)) {
+                entry.classList.remove(removeClass);
+                entry.classList.add(addClass);
+
+                const inputElement = entry.querySelector('[data-element="input"]');
+                
+                if (inputElement) {
+                    $(inputElement).trigger('advanced-shown.kbase');
+                }
+
+            }
+
+            // Also update the count in the paramters.
+            const events = Events.make({ node: container });
+
+            let message,
+                showAdvancedButton;
+
+            if (settings.showAdvanced) {
+                if (advancedInputs.length > 1) {
+                    message = String(advancedInputs.length) + ' advanced parameters showing';
+                } else {
+                    message = String(advancedInputs.length) + ' advanced parameter showing';
+                }
+
+                showAdvancedButton = ui.buildButton({
+                    label: 'hide advanced',
+                    type: 'link',
+                    name: 'advanced-parameters-toggler',
+                    event: {
+                        type: 'toggle-advanced'
+                    },
+                    events: events
+                });
+
+                ui.setContent([areaElement, 'advanced-hidden-message'], '(' + message + ') ' + showAdvancedButton);
+            } else {
+                if (advancedInputs.length > 1) {
+                    message = String(advancedInputs.length) + ' advanced parameters hidden';
+                } else {
+                    message = String(advancedInputs.length) + ' advanced parameter hidden';
+                }
+                showAdvancedButton = ui.buildButton({
+                    label: 'show advanced',
+                    type: 'link',
+                    name: 'advanced-parameters-toggler',
+                    event: {
+                        type: 'toggle-advanced'
+                    },
+                    events: events
+                });
+
+                ui.setContent([areaElement, 'advanced-hidden-message'], '(' + message + ') ' + showAdvancedButton);
+            }
+
+            events.attachEvents();
+        }
+
+        function renderLayout() {
+            const events = Events.make();
+            let formContent = [];
+
+            formContent = formContent.concat([
+                ui.buildPanel({
+                    title: span(['Parameters', span({ dataElement: 'advanced-hidden-message', style: { marginLeft: '6px', fontStyle: 'italic' } })]),
+                    name: 'parameters-area',
+                    body: div({ dataElement: 'parameter-fields' }),
+                    classes: ['kb-panel-light']
```

Comment:
this comment applies wherever there are bits of DOM building going on: could you add a class to each element where there is currently a style, and ideally to all elements so that we have classes there if we need to style things? I started going over these files and had this skeleton for this section:

```js
                    ui.buildPanel({
                        title: span([
                            'Parameters',
                            span({
                                class: 'kb-app-params__advanced-message--parameters',
                                dataElement: 'advanced-hidden-message',
                                style: { marginLeft: '6px', fontStyle: 'italic' }
                            })
                        ]),
                        name: 'parameters-area',
                        body: div({
                            class: 'kb-app-params__fields--parameters',
                            dataElement: 'parameter-fields'
                        }),
                        classes: ['kb-panel-light']
                    })
```